sql/core/src/main/scala/org/apache/spark/sql/ArrowConverters.scala

```diff
@@ -185,18 +225,19 @@ private[sql] abstract class PrimitiveColumnWriter(protected val allocator: BaseA
   protected var count = 0
   protected var nullCount = 0
 
-  override def init(initialSize: Int): Unit = {
+  override def init(initialSize: Option[Int]): this.type = {
+    initialSize.foreach(valueVector.setInitialCapacity)
```

Comment:
I noticed this was never used before, would it increase performance much to set the exact capacity?  Although I'm not sure its possible when using `Iterator[InternalRow]` without making a first pass